### ISSUE TITLE
PAW

### ISSUE THREAD

**Issue by antoine-levitt** [JuliaMolSim/DFTK.jl#987]
I don't want to bother implementing PAW, but I found the very nice
http://staff.ustc.edu.cn/~zqj/posts/VASP-All-Electron-WFC/ so I'm just putting this link up here for reference if somebody is interested.

**Comment by simonganne01** [comment]
I assume nobody is implementing this yet, because it would be very nice since this really is state-of-the-art. I looked at the documentation of gpaw as well, and the things needed to implement this are certainly compatible with DFTK but a lot of work to implement. I want to adopt DFTK as main simulation software but PAW would be a requirement in order to really adopt it

**Comment by mfherbst** [comment]
Indeed @simonganne01 nobody is actively working on PAW. As you mention it would be considerable work (but also very interesting) and we would be open and support any help in this direction.

**Comment by antoine-levitt** [comment]
I don't think it's that much work especially now that we've got numerical pseudos, all the tools are there, it's just a matter of plugging the formulas correctly. The more annoying part is getting that to work with the rest of the stuff (forces, phonons, etc)

**Comment by mfherbst** [comment]
> I don't think it's that much work especially now that we've got numerical pseudos

well we don't have ultrasofts. That would be the first step, getting all this augmentation charge business in place and get that working reliably within an SCF. The step up from there to PAW is not that large, though, I agree.

**Comment by mfherbst** [comment]
Not quite, it's more of a programming question. The next step to expand DFTK towards PAW on the SCF-side and the diagonalisation would be to implement ultrasoft pseudos. This gets much of the plumbing in place you need for PAW.

**Comment by antoine-levitt** [comment]
Looking at https://www.tcm.phy.cam.ac.uk/castep/documentation/WebHelp/content/modules/castep/thcastepultrapseudo.htm it really doesn't seem that hard : it's just a bunch of extra things in H and S that are low rank operators built from functions given on a radial grid, so very similar to numerical norm conserving pseudos

**Comment by simonganne01** [comment]
I see that DFTK is not using the latest version of PseudoPotentialIO, and that the latest version of pseudopotential IO has already defined several classes for ultrasoft etc. outside of DFTK. Is it intended that this be updated to a newer version of pseudopotentialIO and that the pseudopotential classes from there be used?

**Comment by antoine-levitt** [comment]
Pspio will get us the atomic data we need, but we still have to make use of it in dftk

**Comment by simonganne01** [comment]
yes but we want to use the classes of PseudopotentialIO to store the data and then add different terms to the overlap and hamiltonian depending on the pseudopotential through multiple dispatch right?

**Comment by antoine-levitt** [comment]
I'm not very familiar with the pseudopotentials code but those are API questions. To work on this the priority would be to actually code up the equations (using whatever hack is needed to get the info from the psps to DFTK) and to test them. Then the API stuff is relatively minor and can be worked out more constructively once it's clear what functionality is actually needed.

**Comment by mfherbst** [comment]
I agree and yes the idea was to longterm have the pseudo stuff live in pspio.

**Comment by Technici4n** [comment]
Related topic, ultrasoft: #1092.
